influent-java/src/main/java/influent/forward/NioForwardConnection.java

```diff
@@ -114,6 +161,10 @@
   public void onWritable(final SelectionKey key) {
     if (sendResponses()) {
       eventLoop.disableInterestSet(key, SelectionKey.OP_WRITE);
+      if (state == ConnectionState.HELO) {
+        state = ConnectionState.PINGPONG;
+        eventLoop.enableInterestSet(key, SelectionKey.OP_READ);
+      }
```

Comment:
Please add a comment `TODO: disconnect after writing failed PONG`.
The specification says that `Server MUST disconnect the connection if authentication/authorization operation fails.`.